src/components/ComputeModal.js

```diff
@@ -1468,18 +1468,18 @@ export const ComputeModalBase = ({
               'This change requires rebuilding your cloud environment, which will ',
               span({ style: { fontWeight: 600 } }, ['delete all files on built-in hard disk.'])
             ]),
-            existingRuntime.tool === 'RStudio' ? h(SaveFilesHelpRStudio) : h(SaveFilesHelp)
+            existingRuntime?.tool === 'RStudio' ? h(SaveFilesHelpRStudio) : h(SaveFilesHelp)
           ])],
           [willDeletePersistentDisk(), () => h(Fragment, [
             p([
               'To reduce the size of the PD, the existing PD will be deleted and a new one will be created and attached to your virtual machine instance. This will ',
               span({ style: { fontWeight: 600 } }, ['delete all files on the disk.'])
             ]),
-            existingRuntime.tool === 'RStudio' ? h(SaveFilesHelpRStudio) : h(SaveFilesHelp)
+            existingRuntime?.tool === 'RStudio' ? h(SaveFilesHelpRStudio) : h(SaveFilesHelp)
           ])],
           [willRequireDowntime(), () => h(Fragment, [
-            existingRuntime.tool !== desiredTool ?
-              p(['By continuing, you will be changing the application of your cloud environment from ', strong([existingRuntime.tool]), ' to ',
+            existingRuntime && existingRuntime.tool !== desiredTool ?
```

Comment:
Could this use optional chaining like the other `existingRuntime.tool` conditions?

Reply:
I thought about it, but i like that the others have a sensible defaults if there is no existing runtime